statistics/feedback.go

```diff
@@ -425,31 +447,18 @@ func mergeBuckets(bkts []bucket, isNewBuckets []bool, totalCount float64) []buck
 
 func splitBuckets(h *Histogram, feedback *QueryFeedback) ([]bucket, []bool, int64) {
 	bktID2FB, fbNum := buildBucketFeedback(h, feedback)
-	counts := make([]int64, 0, h.Len())
-	for i := 0; i < h.Len(); i++ {
-		bkt, ok := bktID2FB[i]
-		if !ok {
-			counts = append(counts, h.bucketCount(i))
-		} else {
-			counts = append(counts, bkt.getBucketCount(float64(h.bucketCount(i))))
-		}
-	}
-	totCount := int64(0)
-	for _, count := range counts {
-		totCount += count
-	}
 	buckets := make([]bucket, 0, h.Len())
 	isNewBuckets := make([]bool, 0, h.Len())
 	splitCount := getSplitCount(fbNum, defaultBucketCount-h.Len())
 	for i := 0; i < h.Len(); i++ {
 		bkt, ok := bktID2FB[i]
 		// No feedback, just use the original one.
 		if !ok {
-			buckets = append(buckets, bucket{h.GetLower(i), h.GetUpper(i), counts[i], h.Buckets[i].Repeat})
+			buckets = append(buckets, bucket{h.GetLower(i), h.GetUpper(i), h.bucketCount(i), h.Buckets[i].Repeat})
 			isNewBuckets = append(isNewBuckets, false)
 			continue
 		}
-		bkts := bkt.splitBucket(splitCount*len(bkt.feedback)/fbNum, float64(totCount), float64(counts[i]))
+		bkts := bkt.splitBucket(splitCount*len(bkt.feedback)/fbNum, h.totalRowCount(), float64(h.bucketCount(i)))
```

Comment:
Need a dedicated variable for `splitCount*len(bkt.feedback)/fbNum` and add a comment to explain why.